### PR TITLE
feat(appset): reduce cluster secret logging

### PR DESCRIPTION
**Pull request by rumstead**
Currently, the cluster secret matching logs a line every time a cluster generator matches a cluster secret. When the reconciles were happening synchronously the logs had more meaning. Now that reconciles can occur asynchronously, you cannot tie the log line to which applicationset generator caused it.  

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Comment by rumstead** [comment]
```
time="2024-12-09T13:32:08-05:00" level=info msg="listed ApplicationSets" count=1 type=createSecretEventHandler
time="2024-12-09T13:32:11-05:00" level=debug msg="clusters matching labels: 2" applicationset=cluster
time="2024-12-09T13:32:11-05:00" level=debug msg="Using flat mode = false for cluster generator" applicationset=cluster
time="2024-12-09T13:32:11-05:00" level=debug msg="matched cluster secret" applicationset=cluster cluster=cluster-k3d-admin-serverlb-2295321533
time="2024-12-09T13:32:11-05:00" level=debug msg="matched cluster secret" applicationset=cluster cluster=cluster-k3d-dev-serverlb-422902893
time="2024-12-09T13:32:11-05:00" level=info msg="generated 2 applications" applicationset=argocd/cluster generator="{nil &ClusterGenerator{Selector:{map[kubernetes.cnp.io/cluster.segment:multitenant] []},Template:ApplicationSetTemplate{ApplicationSetTemplateMeta:ApplicationSetTemplateMeta{Name:,Namespace:,Labels:map[string]string{},Annotations:map[string]string{},Finalizers:[],},Spec:ApplicationSpec{Source:nil,Destination:ApplicationDestination{Server:,Namespace:,Name:,},Project:,SyncPolicy:nil,IgnoreDifferences:[]ResourceIgnoreDifferences{},Info:[]Info{},RevisionHistoryLimit:nil,Sources:[]ApplicationSource{},},},Values:map[string]string{},FlatList:false,} nil nil nil nil nil nil nil nil}"
time="2024-12-09T13:32:11-05:00" level=debug msg="apps from generator: [{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:cluster-generator-admin GenerateName: Namespace:argocd SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[app-type:cluster] Annotations:map[] OwnerReferences:[] Finalizers:[resources-finalizer.argocd.argoproj.io] ManagedFields:[]} Spec:{Source:&ApplicationSource{RepoURL:https://github.com/rumstead/argocd-playground.git,Path:manifests/kustomize/,TargetRevision:HEAD,Helm:nil,Kustomize:nil,Directory:nil,Plugin:nil,Chart:,Ref:,Name:,} Destination:{Server: Namespace:cluster-generator Name:in-cluster isServerInferred:false} Project:default SyncPolicy:&SyncPolicy{Automated:&SyncPolicyAutomated{Prune:true,SelfHeal:false,AllowEmpty:false,},SyncOptions:[CreateNamespace=true],Retry:nil,ManagedNamespaceMetadata:nil,} IgnoreDifferences:[] Info:[] RevisionHistoryLimit:<nil> Sources:[]} Status:{Resources:[] Sync:{Status: ComparedTo:{Source:{RepoURL: Path: TargetRevision: Helm:nil Kustomize:nil Directory:nil Plugin:nil Chart: Ref: Name:} Destination:{Server: Namespace: Name: isServerInferred:false} Sources:[] IgnoreDifferences:[]} Revision: Revisions:[]} Health:{Status: Message: LastTransitionTime:<nil>} History:[] Conditions:[] ReconciledAt:<nil> OperationState:nil ObservedAt:<nil> SourceType: Summary:{ExternalURLs:[] Images:[]} ResourceHealthSource: SourceTypes:[] ControllerNamespace:} Operation:nil} {TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:cluster-generator-dev GenerateName: Namespace:argocd SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[app-type:cluster] Annotations:map[] OwnerReferences:[] Finalizers:[resources-finalizer.argocd.argoproj.io] ManagedFields:[]} Spec:{Source:&ApplicationSource{RepoURL:https://github.com/rumstead/argocd-playground.git,Path:manifests/kustomize/,TargetRevision:HEAD,Helm:nil,Kustomize:nil,Directory:nil,Plugin:nil,Chart:,Ref:,Name:,} Destination:{Server: Namespace:cluster-generator Name:in-cluster isServerInferred:false} Project:default SyncPolicy:&SyncPolicy{Automated:&SyncPolicyAutomated{Prune:true,SelfHeal:false,AllowEmpty:false,},SyncOptions:[CreateNamespace=true],Retry:nil,ManagedNamespaceMetadata:nil,} IgnoreDifferences:[] Info:[] RevisionHistoryLimit:<nil> Sources:[]} Status:{Resources:[] Sync:{Status: ComparedTo:{Source:{RepoURL: Path: TargetRevision: Helm:nil Kustomize:nil Directory:nil Plugin:nil Chart: Ref: Name:} Destination:{Server: Namespace: Name: isServerInferred:false} Sources:[] IgnoreDifferences:[]} Revision: Revisions:[]} Health:{Status: Message: LastTransitionTime:<nil>} History:[] Conditions:[] ReconciledAt:<nil> OperationState:nil ObservedAt:<nil> SourceType: Summary:{ExternalURLs:[] Images:[]} ResourceHealthSource: SourceTypes:[] ControllerNamespace:} Operation:nil}]" applicationset=argocd/cluster generator="{nil &ClusterGenerator{Selector:{map[kubernetes.cnp.io/cluster.segment:multitenant] []},Template:ApplicationSetTemplate{ApplicationSetTemplateMeta:ApplicationSetTemplateMeta{Name:,Namespace:,Labels:map[string]string{},Annotations:map[string]string{},Finalizers:[],},Spec:ApplicationSpec{Source:nil,Destination:ApplicationDestination{Server:,Namespace:,Name:,},Project:,SyncPolicy:nil,IgnoreDifferences:[]ResourceIgnoreDifferences{},Info:[]Info{},RevisionHistoryLimit:nil,Sources:[]ApplicationSource{},},},Values:map[string]string{},FlatList:false,} nil nil nil nil nil nil nil nil}"
time="2024-12-09T13:32:11-05:00" level=debug msg="unchanged Application" app=argocd/cluster-generator-admin applicationset=argocd/cluster
time="2024-12-09T13:32:11-05:00" level=debug msg="unchanged Application" app=argocd/cluster-generator-dev applicationset=argocd/cluster
time="2024-12-09T13:32:11-05:00" level=info msg="end reconcile in 2.973307334s" applicationset=argocd/cluster requeueAfter=0s
time="2024-12-09T13:32:12-05:00" level=debug msg="clusters matching labels: 2" applicationset=cluster
time="2024-12-09T13:32:12-05:00" level=debug msg="Using flat mode = false for cluster generator" applicationset=cluster
time="2024-12-09T13:32:12-05:00" level=debug msg="matched cluster secret" applicationset=cluster cluster=cluster-k3d-admin-serverlb-2295321533
time="2024-12-09T13:32:12-05:00" level=debug msg="matched cluster secret" applicationset=cluster cluster=cluster-k3d-dev-serverlb-422902893
time="2024-12-09T13:32:12-05:00" level=info msg="generated 2 applications" applicationset=argocd/cluster generator="{nil &ClusterGenerator{Selector:{map[kubernetes.cnp.io/cluster.segment:multitenant] []},Template:ApplicationSetTemplate{ApplicationSetTemplateMeta:ApplicationSetTemplateMeta{Name:,Namespace:,Labels:map[string]string{},Annotations:map[string]string{},Finalizers:[],},Spec:ApplicationSpec{Source:nil,Destination:ApplicationDestination{Server:,Namespace:,Name:,},Project:,SyncPolicy:nil,IgnoreDifferences:[]ResourceIgnoreDifferences{},Info:[]Info{},RevisionHistoryLimit:nil,Sources:[]ApplicationSource{},},},Values:map[string]string{},FlatList:false,} nil nil nil nil nil nil nil nil}"
time="2024-12-09T13:32:12-05:00" level=debug msg="apps from generator: [{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:cluster-generator-admin GenerateName: Namespace:argocd SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[app-type:cluster] Annotations:map[] OwnerReferences:[] Finalizers:[resources-finalizer.argocd.argoproj.io] ManagedFields:[]} Spec:{Source:&ApplicationSource{RepoURL:https://github.com/rumstead/argocd-playground.git,Path:manifests/kustomize/,TargetRevision:HEAD,Helm:nil,Kustomize:nil,Directory:nil,Plugin:nil,Chart:,Ref:,Name:,} Destination:{Server: Namespace:cluster-generator Name:in-cluster isServerInferred:false} Project:default SyncPolicy:&SyncPolicy{Automated:&SyncPolicyAutomated{Prune:true,SelfHeal:false,AllowEmpty:false,},SyncOptions:[CreateNamespace=true],Retry:nil,ManagedNamespaceMetadata:nil,} IgnoreDifferences:[] Info:[] RevisionHistoryLimit:<nil> Sources:[]} Status:{Resources:[] Sync:{Status: ComparedTo:{Source:{RepoURL: Path: TargetRevision: Helm:nil Kustomize:nil Directory:nil Plugin:nil Chart: Ref: Name:} Destination:{Server: Namespace: Name: isServerInferred:false} Sources:[] IgnoreDifferences:[]} Revision: Revisions:[]} Health:{Status: Message: LastTransitionTime:<nil>} History:[] Conditions:[] ReconciledAt:<nil> OperationState:nil ObservedAt:<nil> SourceType: Summary:{ExternalURLs:[] Images:[]} ResourceHealthSource: SourceTypes:[] ControllerNamespace:} Operation:nil} {TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:cluster-generator-dev GenerateName: Namespace:argocd SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[app-type:cluster] Annotations:map[] OwnerReferences:[] Finalizers:[resources-finalizer.argocd.argoproj.io] ManagedFields:[]} Spec:{Source:&ApplicationSource{RepoURL:https://github.com/rumstead/argocd-playground.git,Path:manifests/kustomize/,TargetRevision:HEAD,Helm:nil,Kustomize:nil,Directory:nil,Plugin:nil,Chart:,Ref:,Name:,} Destination:{Server: Namespace:cluster-generator Name:in-cluster isServerInferred:false} Project:default SyncPolicy:&SyncPolicy{Automated:&SyncPolicyAutomated{Prune:true,SelfHeal:false,AllowEmpty:false,},SyncOptions:[CreateNamespace=true],Retry:nil,ManagedNamespaceMetadata:nil,} IgnoreDifferences:[] Info:[] RevisionHistoryLimit:<nil> Sources:[]} Status:{Resources:[] Sync:{Status: ComparedTo:{Source:{RepoURL: Path: TargetRevision: Helm:nil Kustomize:nil Directory:nil Plugin:nil Chart: Ref: Name:} Destination:{Server: Namespace: Name: isServerInferred:false} Sources:[] IgnoreDifferences:[]} Revision: Revisions:[]} Health:{Status: Message: LastTransitionTime:<nil>} History:[] Conditions:[] ReconciledAt:<nil> OperationState:nil ObservedAt:<nil> SourceType: Summary:{ExternalURLs:[] Images:[]} ResourceHealthSource: SourceTypes:[] ControllerNamespace:} Operation:nil}]" applicationset=argocd/cluster generator="{nil &ClusterGenerator{Selector:{map[kubernetes.cnp.io/cluster.segment:multitenant] []},Template:ApplicationSetTemplate{ApplicationSetTemplateMeta:ApplicationSetTemplateMeta{Name:,Namespace:,Labels:map[string]string{},Annotations:map[string]string{},Finalizers:[],},Spec:ApplicationSpec{Source:nil,Destination:ApplicationDestination{Server:,Namespace:,Name:,},Project:,SyncPolicy:nil,IgnoreDifferences:[]ResourceIgnoreDifferences{},Info:[]Info{},RevisionHistoryLimit:nil,Sources:[]ApplicationSource{},},},Values:map[string]string{},FlatList:false,} nil nil nil nil nil nil nil nil}"
time="2024-12-09T13:32:12-05:00" level=debug msg="unchanged Application" app=argocd/cluster-generator-admin applicationset=argocd/cluster
time="2024-12-09T13:32:12-05:00" level=debug msg="unchanged Application" app=argocd/cluster-generator-dev applicationset=argocd/cluster
time="2024-12-09T13:32:12-05:00" level=info msg="end reconcile in 693.567125ms" applicationset=argocd/cluster requeueAfter=0s
```

**Comment by rumstead** [comment]
I would really like to also remove the generator dump from this [Info](https://github.com/argoproj/argo-cd/blob/96d0226a4963d9639aea81ec1d3a310fed390133/applicationset/controllers/template/template.go#L73) line as well.

File: applicationset/generators/cluster.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ func (g *ClusterGenerator) GetTemplate(appSetGenerator *argoappsetv1alpha1.Appli
 }
 
 func (g *ClusterGenerator) GenerateParams(appSetGenerator *argoappsetv1alpha1.ApplicationSetGenerator, appSet *argoappsetv1alpha1.ApplicationSet, _ client.Client) ([]map[string]interface{}, error) {
+	logCtx := log.WithField("applicationset", appSet.GetName())
```
Comment (crenshaw-dev) [inline applicationset/generators/cluster.go:60]: Maybe add the appset namespace since we now allow appsets in any namespace

**Comment by rumstead** [comment]
Debug enabled
```
time="2024-12-09T14:29:33-05:00" level=info msg="listed ApplicationSets" count=1 type=createSecretEventHandler
time="2024-12-09T14:29:33-05:00" level=info msg="processing event for cluster secret" name=cluster-k3d-dev-serverlb-422902893 namespace=argocd type=createSecretEventHandler
time="2024-12-09T14:29:33-05:00" level=info msg="listed ApplicationSets" count=1 type=createSecretEventHandler
time="2024-12-09T14:29:33-05:00" level=debug msg="clusters matching labels: 2" applicationset=cluster namespace=argocd
time="2024-12-09T14:29:33-05:00" level=debug msg="Using flat mode = false for cluster generator" applicationset=cluster namespace=argocd
time="2024-12-09T14:29:33-05:00" level=debug msg="matched cluster secret" applicationset=cluster cluster=cluster-k3d-admin-serverlb-2295321533 namespace=argocd
time="2024-12-09T14:29:33-05:00" level=debug msg="matched cluster secret" applicationset=cluster cluster=cluster-k3d-dev-serverlb-422902893 namespace=argocd
time="2024-12-09T14:29:33-05:00" level=debug msg="apps from generator: [{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:cluster-generator-admin GenerateName: Namespace:argocd SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[app-type:cluster] Annotations:map[] OwnerReferences:[] Finalizers:[resources-finalizer.argocd.argoproj.io] ManagedFields:[]} Spec:{Source:&ApplicationSource{RepoURL:https://github.com/rumstead/argocd-playground.git,Path:manifests/kustomize/,TargetRevision:HEAD,Helm:nil,Kustomize:nil,Directory:nil,Plugin:nil,Chart:,Ref:,Name:,} Destination:{Server: Namespace:cluster-generator Name:in-cluster isServerInferred:false} Project:default SyncPolicy:&SyncPolicy{Automated:&SyncPolicyAutomated{Prune:true,SelfHeal:false,AllowEmpty:false,},SyncOptions:[CreateNamespace=true],Retry:nil,ManagedNamespaceMetadata:nil,} IgnoreDifferences:[] Info:[] RevisionHistoryLimit:<nil> Sources:[]} Status:{Resources:[] Sync:{Status: ComparedTo:{Source:{RepoURL: Path: TargetRevision: Helm:nil Kustomize:nil Directory:nil Plugin:nil Chart: Ref: Name:} Destination:{Server: Namespace: Name: isServerInferred:false} Sources:[] IgnoreDifferences:[]} Revision: Revisions:[]} Health:{Status: Message: LastTransitionTime:<nil>} History:[] Conditions:[] ReconciledAt:<nil> OperationState:nil ObservedAt:<nil> SourceType: Summary:{ExternalURLs:[] Images:[]} ResourceHealthSource: SourceTypes:[] ControllerNamespace:} Operation:nil} {TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:cluster-generator-dev GenerateName: Namespace:argocd SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[app-type:cluster] Annotations:map[] OwnerReferences:[] Finalizers:[resources-finalizer.argocd.argoproj.io] ManagedFields:[]} Spec:{Source:&ApplicationSource{RepoURL:https://github.com/rumstead/argocd-playground.git,Path:manifests/kustomize/,TargetRevision:HEAD,Helm:nil,Kustomize:nil,Directory:nil,Plugin:nil,Chart:,Ref:,Name:,} Destination:{Server: Namespace:cluster-generator Name:in-cluster isServerInferred:false} Project:default SyncPolicy:&SyncPolicy{Automated:&SyncPolicyAutomated{Prune:true,SelfHeal:false,AllowEmpty:false,},SyncOptions:[CreateNamespace=true],Retry:nil,ManagedNamespaceMetadata:nil,} IgnoreDifferences:[] Info:[] RevisionHistoryLimit:<nil> Sources:[]} Status:{Resources:[] Sync:{Status: ComparedTo:{Source:{RepoURL: Path: TargetRevision: Helm:nil Kustomize:nil Directory:nil Plugin:nil Chart: Ref: Name:} Destination:{Server: Namespace: Name: isServerInferred:false} Sources:[] IgnoreDifferences:[]} Revision: Revisions:[]} Health:{Status: Message: LastTransitionTime:<nil>} History:[] Conditions:[] ReconciledAt:<nil> OperationState:nil ObservedAt:<nil> SourceType: Summary:{ExternalURLs:[] Images:[]} ResourceHealthSource: SourceTypes:[] ControllerNamespace:} Operation:nil}]" applicationset=argocd/cluster generator="{nil &ClusterGenerator{Selector:{map[kubernetes.cnp.io/cluster.segment:multitenant] []},Template:ApplicationSetTemplate{ApplicationSetTemplateMeta:ApplicationSetTemplateMeta{Name:,Namespace:,Labels:map[string]string{},Annotations:map[string]string{},Finalizers:[],},Spec:ApplicationSpec{Source:nil,Destination:ApplicationDestination{Server:,Namespace:,Name:,},Project:,SyncPolicy:nil,IgnoreDifferences:[]ResourceIgnoreDifferences{},Info:[]Info{},RevisionHistoryLimit:nil,Sources:[]ApplicationSource{},},},Values:map[string]string{},FlatList:false,} nil nil nil nil nil nil nil nil}"
time="2024-12-09T14:29:33-05:00" level=debug msg="unchanged Application" app=argocd/cluster-generator-admin applicationset=argocd/cluster
time="2024-12-09T14:29:33-05:00" level=debug msg="unchanged Application" app=argocd/cluster-generator-dev applicationset=argocd/cluster
time="2024-12-09T14:29:33-05:00" level=info msg="end reconcile in 158.934ms" applicationset=argocd/cluster requeueAfter=0s
time="2024-12-09T14:29:33-05:00" level=debug msg="clusters matching labels: 2" applicationset=cluster namespace=argocd
time="2024-12-09T14:29:33-05:00" level=debug msg="Using flat mode = false for cluster generator" applicationset=cluster namespace=argocd
time="2024-12-09T14:29:33-05:00" level=debug msg="matched cluster secret" applicationset=cluster cluster=cluster-k3d-admin-serverlb-2295321533 namespace=argocd
time="2024-12-09T14:29:33-05:00" level=debug msg="matched cluster secret" applicationset=cluster cluster=cluster-k3d-dev-serverlb-422902893 namespace=argocd
time="2024-12-09T14:29:33-05:00" level=debug msg="apps from generator: [{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:cluster-generator-admin GenerateName: Namespace:argocd SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[app-type:cluster] Annotations:map[] OwnerReferences:[] Finalizers:[resources-finalizer.argocd.argoproj.io] ManagedFields:[]} Spec:{Source:&ApplicationSource{RepoURL:https://github.com/rumstead/argocd-playground.git,Path:manifests/kustomize/,TargetRevision:HEAD,Helm:nil,Kustomize:nil,Directory:nil,Plugin:nil,Chart:,Ref:,Name:,} Destination:{Server: Namespace:cluster-generator Name:in-cluster isServerInferred:false} Project:default SyncPolicy:&SyncPolicy{Automated:&SyncPolicyAutomated{Prune:true,SelfHeal:false,AllowEmpty:false,},SyncOptions:[CreateNamespace=true],Retry:nil,ManagedNamespaceMetadata:nil,} IgnoreDifferences:[] Info:[] RevisionHistoryLimit:<nil> Sources:[]} Status:{Resources:[] Sync:{Status: ComparedTo:{Source:{RepoURL: Path: TargetRevision: Helm:nil Kustomize:nil Directory:nil Plugin:nil Chart: Ref: Name:} Destination:{Server: Namespace: Name: isServerInferred:false} Sources:[] IgnoreDifferences:[]} Revision: Revisions:[]} Health:{Status: Message: LastTransitionTime:<nil>} History:[] Conditions:[] ReconciledAt:<nil> OperationState:nil ObservedAt:<nil> SourceType: Summary:{ExternalURLs:[] Images:[]} ResourceHealthSource: SourceTypes:[] ControllerNamespace:} Operation:nil} {TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:cluster-generator-dev GenerateName: Namespace:argocd SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[app-type:cluster] Annotations:map[] OwnerReferences:[] Finalizers:[resources-finalizer.argocd.argoproj.io] ManagedFields:[]} Spec:{Source:&ApplicationSource{RepoURL:https://github.com/rumstead/argocd-playground.git,Path:manifests/kustomize/,TargetRevision:HEAD,Helm:nil,Kustomize:nil,Directory:nil,Plugin:nil,Chart:,Ref:,Name:,} Destination:{Server: Namespace:cluster-generator Name:in-cluster isServerInferred:false} Project:default SyncPolicy:&SyncPolicy{Automated:&SyncPolicyAutomated{Prune:true,SelfHeal:false,AllowEmpty:false,},SyncOptions:[CreateNamespace=true],Retry:nil,ManagedNamespaceMetadata:nil,} IgnoreDifferences:[] Info:[] RevisionHistoryLimit:<nil> Sources:[]} Status:{Resources:[] Sync:{Status: ComparedTo:{Source:{RepoURL: Path: TargetRevision: Helm:nil Kustomize:nil Directory:nil Plugin:nil Chart: Ref: Name:} Destination:{Server: Namespace: Name: isServerInferred:false} Sources:[] IgnoreDifferences:[]} Revision: Revisions:[]} Health:{Status: Message: LastTransitionTime:<nil>} History:[] Conditions:[] ReconciledAt:<nil> OperationState:nil ObservedAt:<nil> SourceType: Summary:{ExternalURLs:[] Images:[]} ResourceHealthSource: SourceTypes:[] ControllerNamespace:} Operation:nil}]" applicationset=argocd/cluster generator="{nil &ClusterGenerator{Selector:{map[kubernetes.cnp.io/cluster.segment:multitenant] []},Template:ApplicationSetTemplate{ApplicationSetTemplateMeta:ApplicationSetTemplateMeta{Name:,Namespace:,Labels:map[string]string{},Annotations:map[string]string{},Finalizers:[],},Spec:ApplicationSpec{Source:nil,Destination:ApplicationDestination{Server:,Namespace:,Name:,},Project:,SyncPolicy:nil,IgnoreDifferences:[]ResourceIgnoreDifferences{},Info:[]Info{},RevisionHistoryLimit:nil,Sources:[]ApplicationSource{},},},Values:map[string]string{},FlatList:false,} nil nil nil nil nil nil nil nil}"
time="2024-12-09T14:29:33-05:00" level=debug msg="unchanged Application" app=argocd/cluster-generator-admin applicationset=argocd/cluster
time="2024-12-09T14:29:33-05:00" level=debug msg="unchanged Application" app=argocd/cluster-generator-dev applicationset=argocd/cluster
time="2024-12-09T14:29:33-05:00" level=info msg="end reconcile in 37.093792ms" applicationset=argocd/cluster requeueAfter=0s
time="2024-12-09T14:30:21-05:00" level=debug msg="received update event from owning an application" app=argocd/cluster-generator-admin
time="2024-12-09T14:30:21-05:00" level=debug msg="requeue: false caused by application cluster-generator-admin" app=argocd/cluster-generator-admin requeue=false
time="2024-12-09T14:30:22-05:00" level=debug msg="received update event from owning an application" app=argocd/cluster-generator-dev
time="2024-12-09T14:30:22-05:00" level=debug msg="requeue: false caused by application cluster-generator-dev" app=argocd/cluster-generator-dev requeue=false
```

Info level
```
time="2024-12-09T14:32:58-05:00" level=info msg="processing event for cluster secret" name=cluster-k3d-admin-serverlb-2295321533 namespace=argocd type=createSecretEventHandler
time="2024-12-09T14:32:58-05:00" level=info msg="Starting workers" controller=applicationset controllerGroup=argoproj.io controllerKind=ApplicationSet worker count=10
time="2024-12-09T14:32:58-05:00" level=info msg="listed ApplicationSets" count=1 type=createSecretEventHandler
time="2024-12-09T14:32:58-05:00" level=info msg="processing event for cluster secret" name=cluster-k3d-dev-serverlb-422902893 namespace=argocd type=createSecretEventHandler
time="2024-12-09T14:32:58-05:00" level=info msg="listed ApplicationSets" count=1 type=createSecretEventHandler
time="2024-12-09T14:32:58-05:00" level=info msg="generated 2 applications" applicationset=argocd/cluster
time="2024-12-09T14:32:58-05:00" level=info msg="end reconcile in 151.119917ms" applicationset=argocd/cluster requeueAfter=0s
time="2024-12-09T14:32:58-05:00" level=info msg="generated 2 applications" applicationset=argocd/cluster
time="2024-12-09T14:32:58-05:00" level=info msg="end reconcile in 22.246166ms" applicationset=argocd/cluster requeueAfter=0s
```

File: applicationset/controllers/template/template.go [hunk 2]
```diff
@@ -69,9 +69,11 @@ func GenerateApplications(logCtx *log.Entry, applicationSetInfo argov1alpha1.App
 				res = append(res, *app)
 			}
 		}
-
-		logCtx.WithField("generator", requestedGenerator).Infof("generated %d applications", len(res))
```
Comment (rumstead) [inline applicationset/controllers/template/template.go:73]: Based on the convo in slack, I am removing the struct dump for info level logs.

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**